src/Simulation/QCTraceSimulator/SortedQubitPool.cs

```diff
@@ -0,0 +1,321 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime {
+
+    /// <summary>
+    /// Time to track qubit availability (DepthTime). Time is also used to track dependencies:
+    /// Gates that happen later on a qubit cannot be scheduled before gates on the same qubit.
+    /// This is tracked transitively via time, but zero gate depth prevents this from happening.
+    /// This is solved by having additional time component - DependencyTime.
+    /// </summary>
+    internal struct ComplexTime
+    {
+        /// <summary>
+        /// Time used to track qubit availability time. Double to match configuration interface.
+        /// </summary>
+        internal double DepthTime { get; private set; }
+
+        /// <summary>
+        /// This is only used for comparisons when DepthTime is equal. This time tracks dependencies.
```

Comment:
It's not clear what it means to "track dependencies". Dependencies between what? What kind of dependencies?

Reply:
Updated wording will be in the next push.